### ISSUE TITLE
Move from stackD3V3 to d3.stack v4

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#1375]
It's a significantly different API but it provides the same or better functionality.

https://github.com/d3/d3-shape/blob/master/README.md#stack

https://github.com/d3/d3/blob/master/CHANGES.md#shapes-d3-shape

**Comment by gordonwoodhull** [comment]
As a first step, I'm leaving all the existing stack infrastructure (layer objects) in place, transforming the data on input to d3v4 stack, and reading it back out into the layer objects after layout is done.

This works fine and doesn't break any tests. At first glance, it looks like streamgraphs do work out of the box, although they would need dots & lines at the bottom as well:

<img width="749" alt="image" src="https://user-images.githubusercontent.com/1366709/39032050-cdea0bc0-4439-11e8-978d-f2773e9d88f2.png">

```js
  chart
      .stackLayout(d3.stack()
                   .order(d3.stackOrderInsideOut)
                   .offset(d3.stackOffsetWiggle))
```

Unclear to me if there is any benefit to moving closer to the d3v4 paradigm on this - since it doesn't have any stack/layer objects, that would be a big change.

**Comment by gordonwoodhull** [comment]
Released in 3.0 alpha 12
